subworkflows/align_short/subworkflows/local/align_short.nf

```diff
@@ -0,0 +1,38 @@
+//
+// Align short read (HiC and Illumina) data against the genome
+//
+
+include { SAMTOOLS_FASTQ } from '../../modules/nf-core/modules/samtools/fastq/main'
+include { BWAMEM2_MEM    } from '../../modules/nf-core/modules/bwamem2/mem/main'
+include { MARKDUP_STATS  } from '../../subworkflows/local/markdup_stats'
+
+workflow ALIGN_SHORT {
+    take:
+    fasta // channel: /path/to/fasta
+    index // channel: [ val(meta), /path/to/bwamem2/]
+    reads // channel: [ val(meta), [ datafile ] ]
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Convert from CRAM to FASTQ
+    SAMTOOLS_FASTQ ( reads )
+    ch_versions = ch_versions.mix(SAMTOOLS_FASTQ.out.versions.first())
+
+    // Align Fastq to Genome
+    BWAMEM2_MEM ( SAMTOOLS_FASTQ.out.fastq, index, [] )
+    ch_versions = ch_versions.mix(BWAMEM2_MEM.out.versions.first())
+
+    // Merge, markdup, convert, and stats
+    MARKDUP_STATS ( BWAMEM2_MEM.out.bam, fasta )
+    ch_versions = ch_versions.mix(MARKDUP_STATS.out.versions)
```

Comment:
I would remove `first()` from everything then. In [`bwamem2/mem`](https://github.com/nf-core/modules/blob/master/modules/bwamem2/mem/main.nf), `first()` would only return the version for `bwamem2`, not `samtools`.

Reply:
`ch_versions` is actually a channel of paths that collects all the `versions.yml`, so apparently no constraints on the number of versions reported in the yml files

Reply:
From discussions with @muffato:

> `unique()` is there in nf-core pipelines, part of the template for implementing `dumpsoftwareversions`, so leaving it in the workflow.
> 
> `first()` is a mixed one. Seems the best practice would be to add for those modules that maybe be run multiple times or for all modules. A hard no for the subworkflows because then we would only get the versions for the first module.